lib/logflare/backends/adaptor/otlp_adaptor/protobuf_formatter.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Logflare.Backends.Adaptor.OtlpAdaptor.ProtobufFormatter do
+  alias Opentelemetry.Proto.Collector.Logs.V1.ExportLogsServiceRequest
+  alias Opentelemetry.Proto.Collector.Logs.V1.ExportLogsServiceResponse
+  alias Opentelemetry.Proto.Common.V1.ArrayValue
+  alias Opentelemetry.Proto.Common.V1.AnyValue
+  alias Opentelemetry.Proto.Common.V1.InstrumentationScope
+  alias Opentelemetry.Proto.Common.V1.KeyValue
+  alias Opentelemetry.Proto.Common.V1.KeyValueList
+  alias Opentelemetry.Proto.Logs.V1.LogRecord
+  alias Opentelemetry.Proto.Logs.V1.ResourceLogs
+  alias Opentelemetry.Proto.Logs.V1.ScopeLogs
+  alias Opentelemetry.Proto.Logs.V1.SeverityNumber
+  alias Opentelemetry.Proto.Resource.V1.Resource
+  alias Logflare.LogEvent
+
+  @behaviour Tesla.Middleware
+
+  @protobuf_content_type "application/x-protobuf"
+
+  @impl true
+  def call(env, next, opts) do
+    metadata = opts[:metadata] || %{}
+
+    response =
+      env
+      |> Tesla.put_header("content-type", @protobuf_content_type)
+      |> Tesla.put_body(transform_batch(env.body, metadata))
+      |> Tesla.run(next)
+
+    with {:ok, %{status: 200} = env} <- response do
+      if Tesla.get_header(env, "content-type") == @protobuf_content_type do
+        response = Protobuf.decode(env.body, ExportLogsServiceResponse)
+        {:ok, Tesla.put_body(env, response)}
+      else
+        response
+      end
+    end
+  end
+
+  defp transform_batch(events, metadata) do
+    attributes = for {k, v} <- metadata, do: {"logflare.#{k}", v}
+
+    %ExportLogsServiceRequest{
+      resource_logs: [
+        %ResourceLogs{
+          resource: build_resource(attributes),
```

Comment:
can populate the resource with more information as per [here](https://github.com/Logflare/logflare/blob/5c9617da411d757f488368d8969954f332d619db/lib/logflare/backends/user_monitoring.ex#L19)
i think we don't need to set the attribute metadata as of now.